wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WFileWidget.java

```diff
@@ -110,12 +132,86 @@ protected boolean doHandleRequest(final Request request) {
 		boolean changed = value != null || current != null;
```

Comment:
Can `value` be `null`? If not this test can actually be a bit confusing. This is existing code, but would be good to look at and clarify and deal with a `null` value correctly.

Reply:
Below is logic 
```java
boolean changed = value != null || current != null;
```
Test `WFileWidget_Test.testDoHandleRequest`, following situations are covered
1) value != null and current = null -> changed = true
2) value != null and current != null -> changed = true
3) value = null and current != null -> changed = true
4) value = null and current = null -> changed = false